R/drive_request.R

```diff
@@ -44,20 +44,30 @@ build_request <- function(endpoint = character(),
     stop("Endpoint not recognized:\n", endpoint, call. = FALSE)
   }
 
+  out <- list(token = token)
+
+  if (grepl("media", endpoint)) {
+    out$body <- params$body
+    params$body <- NULL
+  }
+
   ## use the spec to vet and rework request parameters
   params <-   match_params(params, ept$parameters)
   params <- handle_repeats(params, ept$parameters)
   params <- check_enums(params, ept$parameters)
   params <- partition_params(params,
                              extract_path_names(ept$path),
                              extract_body_names(ept$parameters))
-  out <- list(
-    method = ept$method,
-    path = paste0("drive/v3/", glue::glue_data(params$path_params, ept$path)),
-    query = c(params$query_params),
-    body = c(params$body_params),
-    token = token
-  )
+
+  out$method <- ept$method
+
+  if (grepl("media", endpoint)) {
+    out$path <- glue::glue_data(params$path_params, ept$path)
+  } else {
+    out$path <-  paste0("drive/v3/", glue::glue_data(params$path_params, ept$path))
+    out$body <- c(params$body_params)
+  }
+  out$query = c(params$query_params)
```

Comment:
As for the body, could you handle that in the small number of functions that upload media? So call `build_request()` in the usual way but edit the body component prior to `make_request()`? I would just like to get all this special logic out of the basic request builder.